---
version: '3.6'
services:
  portainer:
    image: portainer/portainer-ce:${PORTAINER_IMG_VER:-2.13.1}
    container_name: portainer
    restart: unless-stopped
    command: -H tcp://socket-proxy:2375
    volumes:
      - "${PORTAINER_DATA_DIR:-./data}:/data"
    networks:
      - proxy
      - socket-proxy
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=proxy"
      - "traefik.http.routers.portainer-proxy.entrypoints=https"
      - "traefik.http.routers.portainer-proxy.rule=Host(`portainer.${DOMAIN_NAME}`)"
      - "traefik.http.routers.portainer-proxy.tls=true"
      - "traefik.http.routers.portainer-proxy.service=portainer-service"
      - "traefik.http.services.portainer-service.loadbalancer.server.port=${PORTAINER_PORT:-9000}"
      - "traefik.http.routers.portainer-proxy.middlewares=chain-authelia@file"

networks:
  proxy:
    external: true
  socket-proxy:
    external: true
